src/Evolution/Systems/CurvedScalarWave/Worldtube/SingletonActions/ReceiveElementData.hpp

```diff
@@ -66,10 +66,6 @@ struct ReceiveElementData {
         db::get<Tags::ElementFacesGridCoordinates<Dim>>(box).size();
     const auto& time_step_id = db::get<::Tags::TimeStepId>(box);
     auto& inbox = tuples::get<Tags::SphericalHarmonicsInbox<Dim>>(inboxes);
-    if (inbox.count(time_step_id) == 0 or
-        inbox.at(time_step_id).size() < expected_number_of_senders) {
-      return {Parallel::AlgorithmExecution::Retry, std::nullopt};
-    }
```

Comment:
Related to an earlier comment: I think these checks are correct and the others should be omitted.